Automattic/studio
# Add context for sync sites

- Closes https://github.com/Automattic/dotcom-forge/issues/9573

## Proposed Changes

- Add context and move existing management sync hook inside it.
- Improves the performance avoiding to read the appData file each time we change the tab to Sync.
- Adapt tests

## Testing Instructions



Apply this diff to better highlight the loading time for reading connected sites.

```
diff --git a/src/ipc-handlers.ts b/src/ipc-handlers.ts
index 475767c..3b92108 100644
--- a/src/ipc-handlers.ts
+++ b/src/ipc-handlers.ts
@@ -280,6 +280,10 @@ export async function disconnectWpcomSite(
 	return updatedConnections.filter( ( conn ) => conn.localSiteId === localSiteId );
 }
 
+function sleep( ms: number ) {
+	return new Promise( ( resolve ) => setTimeout( resolve, ms ) );
+}
+
 export async function getConnectedWpcomSites(
 	event: IpcMainInvokeEvent,
 	localSiteId: string
@@ -287,6 +291,7 @@ export async function getConnectedWpcomSites(
 	const userData = await loadUserData();
 	const currentUserId = userData.authToken?.id;
 
+	await sleep( 3000 );
 	if ( ! currentUserId ) {
 		return [];
 	}

```

* Run `STUDIO_SITE_SYNC=true npm start`
* Click on the Sync tab.
* Connect a site.
* Switch between tabs.
* Return to the Sync tab.
* Confirm that the connected sites are still displayed.
* Before this PR, the initial screen briefly appeared before the connected sites finished loading.


## Pre-merge Checklist
